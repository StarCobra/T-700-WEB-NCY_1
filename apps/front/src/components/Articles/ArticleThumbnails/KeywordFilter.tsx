import * as React from 'react';
import '../../../style/article.scss'
import { Box, MenuItem, TextField } from '@mui/material';

export default function KeywordFilter() {
    const [keywords, setKeywords] = React.useState<string[]>([])
    console.log(keywords)
    const handleChangeSelectedKeywords = (event: React.ChangeEvent<HTMLInputElement>) => {
    const value = event.target.value
    setKeywords(typeof value === 'string' ? value.split(',') : value)
  }
  return (
    <div className="keywordFilterContainer">
        <div className="selectFilter">
            <Box className="SelectKeywordsBox" color="warning">
            <TextField 
              label="Select a keyword" 
              select 
              className='SelectKeywordsLabel' 
              value={keywords} 
              onChange={handleChangeSelectedKeywords} 
              fullWidth
              SelectProps={{ multiple:true }}
              color='warning'
            >
              <MenuItem value="0">BTC</MenuItem>
              <MenuItem value="1">Bankerupt</MenuItem>
              <MenuItem value="2">Crisis</MenuItem>
            </TextField>
          </Box>
        </div>
    </div>
  )
}
 