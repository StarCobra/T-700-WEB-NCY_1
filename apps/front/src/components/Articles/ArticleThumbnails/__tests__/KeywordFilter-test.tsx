import React from "react";
import 'jsdom-global/register';
import {act, create} from "react-test-renderer";
import KeywordFilter from "../KeywordFilter";

describe("Components > CryptoArray > CryptoArrayDisplay", () => {
    const tests = [
        {
            description: "generic",
            props: {},
        },
    ];
    tests.forEach((test) => {
        it(`should render the component ${test.description}`, async () => {
            let root;
            await act(async () => {
                root = create(<KeywordFilter {...test.props} />);
            });
            expect(root).toMatchSnapshot();
        });
    });
});
