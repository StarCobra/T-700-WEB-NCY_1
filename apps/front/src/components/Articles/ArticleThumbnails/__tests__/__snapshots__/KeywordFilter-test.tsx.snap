// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > CryptoArray > CryptoArrayDisplay should render the component generic 1`] = `
<div
  className="keywordFilterContainer"
>
  <div
    className="selectFilter"
  >
    <div
      className="SelectKeywordsBox MuiBox-root css-4stev7"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root SelectKeywordsLabel css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorWarning MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-mk9tez-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Select a keyword
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorWarning MuiInputBase-fullWidth MuiInputBase-formControl css-nmbvik-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <div
            aria-controls=":r1:"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby=":r0:-label :r0:"
            className="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              â€‹
            </span>
          </div>
          <input
            aria-hidden={true}
            aria-invalid={false}
            autoFocus={false}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Select a keyword
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
